src/core/server/bootstrap.ts

```diff
@@ -81,11 +81,28 @@ export async function bootstrap({ configs, cliArgs, applyConfigOverrides }: Boot
   }
 
   try {
+    const { preboot } = await root.preboot();
+
+    // If setup is on hold then preboot server is supposed to serve user requests and we can let
+    // dev parent process know that we are ready for dev mode.
+    const isSetupOnHold = preboot.isSetupOnHold();
```

Comment:
__note:__ if we don't signal to the parent process (dev mode only) that we're ready to serve user requests it will not let them pass.